src/postgres/commands/copyConnectionString.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ClientConfig } from 'pg';
+import * as vscode from 'vscode';
+import { IActionContext, IParsedError, parseError } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { firewallNotConfiguredErrorType, invalidCredentialsErrorType, PostgresDatabaseTreeItem } from '../tree/PostgresDatabaseTreeItem';
+import { configurePostgresFirewall } from './configurePostgresFirewall';
+import { enterPostgresCredentials } from './enterPostgresCredentials';
+
+export async function copyConnectionString(context: IActionContext, node: PostgresDatabaseTreeItem): Promise<void> {
+    const message = 'The connection string has been copied to the clipboard';
+    if (!node) {
+        node = <PostgresDatabaseTreeItem>await ext.tree.showTreeItemPicker(PostgresDatabaseTreeItem.contextValue, context);
+    }
+
+    let clientConfig: ClientConfig | undefined;
+    while (!clientConfig) {
```

Comment:
You copied this code from a different file. Please put it in a place where the code can be shared